config_parser_test.cc

```diff
@@ -9,3 +12,184 @@ TEST(NginxConfigParserTest, SimpleConfig) {
+  protected:
+    bool ParseString(const std::string config_string) {
+      std::stringstream config_stream(config_string);
+      return parser_.Parse(&config_stream, &out_config_);
+    }
+    NginxConfigParser parser_;
+    NginxConfig out_config_;
+};
+
+// Use TEST_F to use a fixture, helps with not repeating test setup
+TEST_F(NginxStringConfigTest, AnotherSimpleConfig) {
+  EXPECT_TRUE(ParseString("foo bar;"));
+  EXPECT_EQ(1, out_config_.statements_.size())
+    << "Config has one statements";
+  EXPECT_EQ("foo", out_config_.statements_.at(0)->tokens_.at(0));
+}
+
+TEST_F(NginxStringConfigTest, InvalidConfig) {
+  EXPECT_FALSE(ParseString("foo bar"));
+}
+
+TEST_F(NginxStringConfigTest, NestedConfig) {
+  EXPECT_TRUE(ParseString("server {\n\
+                             listen 80;\n\
+                             location ~ ^/forum/(?P.*)$ {\n\
+                               return 301 $scheme://forum.domain.com/$1;\n\
+                             }\n\
+                           }"));
+  // TODO: Test the contents of out_config_;
+}
+
+TEST_F(NginxStringConfigTest, DoubleNestedConfig) {
+  EXPECT_TRUE(ParseString("user  www www;\n\
+                             \n\
+                             events {\n\
+                               worker_connections 4096;\n\
+                             }\n\
+                             \n\
+                             http {\n\
+                               include conf/mime.types;\n\
+                               server {\n\
+                                 listen 80;\n\
+                                 server_name domain1.com www.domain1.com;\n\
+                                 \n\
+                                 location ~ \.php$ {\n\
+                                   fastcgi_pass 127.0.0.1:1025;\n\
+                                 }\n\
+                               }\n\
+                            }"));
+}
+
+TEST_F(NginxStringConfigTest, UnbalancedConfig) {
+  EXPECT_FALSE(ParseString("server { listen 80; "));
+  EXPECT_FALSE(ParseString("server listen 80; }"));
+}
+
+TEST_F(NginxStringConfigTest, EmptyConfig) {
+  EXPECT_FALSE(ParseString(""));
+  EXPECT_FALSE(ParseString("# im the only one in here"));
+  EXPECT_FALSE(ParseString("##\n"));
+}
+
+TEST_F(NginxStringConfigTest, OnlyCurlyConfig) {
+  EXPECT_FALSE(ParseString("{}"));
+}
+
+TEST_F(NginxStringConfigTest, NoBlockNameConfig) {
+  EXPECT_FALSE(ParseString("{ listen 80; }"));
+}
+
+TEST_F(NginxStringConfigTest, CommentConfig) {
+  EXPECT_TRUE(ParseString("# hello world\n server { listen 80; }"));
+  EXPECT_TRUE(ParseString("server {\n\
+                            # hello world\n\
+                            listen 80;\n\
+                          }"));
+  EXPECT_TRUE(ParseString("server { # simple reverse-proxy\n\
+                             listen 80;\n\
+                           }"));
+}
+
+TEST_F(NginxStringConfigTest, FullConfig) {
+  EXPECT_TRUE(ParseString("user  www www;\n\
```

Comment:
[Got it](https://github.com/apfong/nginx-configparser/pull/1/commits/435bd2d8a54da461ee5c0ff4656ec46b3a755ea2)